playerattacktimer/src/main/java/net/runelite/client/plugins/playerattacktimer/PlayerAttackTimerPlugin.java

```diff
@@ -0,0 +1,229 @@
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package net.runelite.client.plugins.playerattacktimer;
+
+import com.google.common.base.Splitter;
+import com.google.inject.Provides;
+import java.util.HashMap;
+import java.util.Map;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import lombok.Getter;
+import net.runelite.api.Client;
+import net.runelite.api.GameState;
+import net.runelite.api.Player;
+import net.runelite.api.events.AnimationChanged;
+import net.runelite.api.events.GameStateChanged;
+import net.runelite.api.events.GameTick;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.eventbus.EventBus;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.client.events.ConfigChanged;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.plugins.PluginType;
+import net.runelite.client.ui.overlay.OverlayManager;
+import org.pf4j.Extension;
+
+@Singleton
+@Extension
+@PluginDescriptor(
+	name = "Player Attack Timer",
+	enabledByDefault = false,
+	description = "Tick delay timer for player attacks.",
+	tags = {"player", "attack", "tick", "timer", "delay"},
+	type = PluginType.UTILITY
+)
+public class PlayerAttackTimerPlugin extends Plugin
+{
+	private static final Map<Integer, Integer> animationTickMap = new HashMap<>();
+
+	static
+	{
+		animationTickMap.put(1658, 4); // Abyssal whip
+	}
+
+	private final Map<Integer, Integer> customAnimationTickMap = new HashMap<>();
+
+	private static final Splitter NEWLINE_SPLITTER = Splitter
+		.on("\n")
+		.omitEmptyStrings()
+		.trimResults();
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private EventBus eventBus;
+
+	@Inject
+	PlayerAttackTimerConfig config;
+
+	@Inject
+	private OverlayManager overlayManager;
+
+	@Inject
+	private PlayerOverlay playerOverlay;
+
+	private boolean enabled;
+
+	@Getter
+	private int ticksUntilNextAnimation;
+
+	@Provides
+	PlayerAttackTimerConfig getConfig(final ConfigManager configManager)
+	{
+		return configManager.getConfig(PlayerAttackTimerConfig.class);
+	}
+
+	@Override
+	protected void startUp()
+	{
+		if (client.getGameState() != GameState.LOGGED_IN)
+		{
+			return;
+		}
+
+		init();
+	}
+
+	@Override
+	protected void shutDown()
+	{
+		eventBus.unregister(this);
+
+		overlayManager.remove(playerOverlay);
+
+		customAnimationTickMap.clear();
+
+		enabled = false;
+	}
+
+	@Subscribe
+	private void onConfigChanged(final ConfigChanged event)
+	{
+		if (!event.getGroup().equals("playerattacktimer"))
```

Comment:
doesnt account for mirror mode compatability, would need to reload plugin for it to take effect currently.